### PR TITLE
[Test] Improve scalability envelope

### PR DESCRIPTION
**Pull request by rkooo567**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by wuisawesome — COMMENTED** [review]
yay can you update the README to 1k nodes too?

**Comment by rkooo567** [comment]
Btw, cc @ericl We verified we could scale up to 1000+ nodes, but we found regression from the many tasks test (scheduling seems to be really slow now).

File: benchmarks/distributed/test_distributed.py [hunk 1]
```diff
@@ -6,11 +6,11 @@
 from time import sleep, perf_counter
 from tqdm import tqdm, trange
 
-TEST_NUM_NODES = 64
+TEST_NUM_NODES = 65
 MAX_ACTORS_IN_CLUSTER = 10000
-MAX_RUNNING_TASKS_IN_CLUSTER = 10000
+MAX_RUNNING_TASKS_IN_CLUSTER = 16000
```
Comment (wuisawesome) [inline benchmarks/distributed/test_distributed.py:11]: We probably shouldn't change this one wince we've never seen it pass.

**Review by wuisawesome — APPROVED** [review]
Let's keep max running tasks in cluster constant, then merge this?

**Pull request merged**